### PR TITLE
Site Profiler: add tooltip to the hosting provider if its popular CDN

### PR DESCRIPTION
**Pull request by ouikhuan**
Related to #82241

## Proposed Changes

* Adding a tooltip if it's the hosting provider is a popular CDN we recognized. If we have support_url from the API, we'll provide links in the popup, otherwise, it'll be text only.
Screen Shot 2023-09-28 at 3 24 09 PM

## Testing Instructions

* Go to http://calypso.localhost:3000/site-profiler
* Type in a domain that's using one of the popular CDNs, you can find one of them from the diff here: D123383-code
* Check if there's an information icon showing next to the hosting provider.
* Click it and see if the tooltips shows up, click again to hide it
* Try to see if there's a `contact [hosting_provider_name]` link in the tooltip like the screenshot above
* Click on the link and see if it brings you to the support page of the hosting provider
* Check other domains and make sure there's no regression for other conditions.
* For now, the hosting provider name will only shown as a link if `home_url` is provided, which is not implemented from the backend yet.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by bogiii — APPROVED** [review]
I've tested it and it works as it is descibed. Well done!

The only comment I have is a feeling that we can move this block of logic into a dedicated component such as a `hosting-provider-name` or `provider-name`.

What do you think?

File: client/site-profiler/components/hosting-information/index.tsx [hunk 1]
```diff
@@ -11,16 +14,75 @@ interface Props {
...
+				{ hostingProvider?.name }
+			</Button>
+		) : (
+			hostingProvider?.name
+		);
+	};
 
 	return (
 		<div className="hosting-information">
 			<h3>Hosting information</h3>
```
Comment (bogiii) [inline client/site-profiler/components/hosting-information/index.tsx]: If you have a chance, please wrap this string with a translate.
Thanks in advance! 🙏

**Comment by javierarce** [comment]
Nice work, @ouikhuan!

Just a couple of comments:

- We should adjust the padding so it's uniform around the content of the tooltip.
- I think we should underline the link and use the same color we're using for the links in the profiler.
- If we use the link as the trigger element for the tooltip, then we won't allow the user to navigate to Cloudflare. I think we could add a small information icon as we do in Calypso:

image

**Comment by ouikhuan** [comment]
> * We should adjust the padding so it's uniform around the content of the tooltip.
> * I think we should underline the link and use the same color we're using for the links in the profiler.
> * If we use the link as the trigger element for the tooltip, then we won't allow the user to navigate to Cloudflare. I think we could add a small information icon as we do in Calypso:

Thanks for the review @javierarce , I've updated the PR based on your suggestions!

**Review by zaerl — APPROVED** [review]
LGTM.

**Comment by javierarce** [comment]
Thanks for the changes, they look good to me. That said, it would be great if we could change the direction of the tooltip to avoid blocking the A records below.

**Review by bogiii — APPROVED** [review]
Works good!

**Comment by ouikhuan** [comment]
> Thanks for the changes, they look good to me. That said, it would be great if we could change the direction of the tooltip to avoid blocking the A records below.

Nice catch, I've change it so it'll showing on top of the icon now 👍

**Review by zaerl — APPROVED** [review]
LGTM.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9057211

Thank you @ouikhuan for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9057211).

**Pull request merged**